cfme/tests/infrastructure/test_vm_rest.py

```diff
@@ -135,3 +138,38 @@ def test_delete_vm_from_collection(vm):
         casecomponent: Infra
     """
     delete_resources_from_collection([vm], not_found=True, num_sec=300, delay=10)
+
+
+@pytest.mark.tier(1)
+@pytest.mark.ignore_stream("5.10")
+@pytest.mark.meta(coverage=[1684681])
+def test_filtering_vm_with_multiple_ips(appliance, provider):
+    """
+    Polarion:
+        assignee: pvala
+        caseimportance: high
+        casecomponent: Rest
+        initialEstimate: 1/4h
+        setup:
+            1. Add a provider.
+        testSteps:
+            1. Select a VM with multiple IP addresses and note one ipaddress.
+            2. Send a GET request with the noted ipaddress.
+                GET /api/vms?expand=resources&attributes=ipaddresses&filter[]=ipaddresses=':ipaddr'
+        expectedResults:
+            1.
+            2. Selected VM must be present in the resources sent by response.
+
+    Bugzilla:
+        1684681
+    """
+    # 1
+    vm = appliance.collections.infra_vms.instantiate(
+        provider.data["cap_and_util"]["capandu_vm"], provider
```

Comment:
@valaparthvi do you think its a good idea to catch a KeyError here and use pytest.skip in case a provider doesn't have required keys?
Or how about: 
`required_fields=[["cap_and_util", "capandu_azone"]],` in pytest mark provider

Reply:
I agree here

Reply:
Hmm, `required_fields` is new to me, surely I can add it, thanks KK, John :)